open-grant-proposals/open-proposal-tinker-drive.md

```diff
@@ -0,0 +1,97 @@
+# Open Grant Proposal: Native file syncing on Filecoin via unixfs-datastore
+
+**Name of Project:** tinker drive - Native file syncing on Filecoin via unixfs-datastore
+
+**Proposal Category:** `core-dev` / `app-dev`
+
+**Proposer:** [@requilence](https://github.com/requilence)
+
+**Do you agree to open source all work you do on behalf of this RFP and dual-license under MIT and APACHE2 licenses?:** Yes
+
+# Project Description
+
+Native file syncing and sharing is an essential feature to provide competitive user experience vs. cloud solutions. 
+Current solutions that allow to mount IPFS with FUSE are suboptimal and don’t provide fluent user experience with the speed of a native filesystem, like Dropbox does.
+
+We propose to develop a cross platform go library to enable full OS integration (ipfs/filecoin with native file system). It will enable working with the same versions of files on different devices with different software. Example: a user can edit an xls file on her computer and it will be stored in Filecoin immediately without data duplication on the client device. Files will be fully encrypted when stored in IPFS/Filecoin.
+
+## Value
+
+Allowing end-users to back up files from their computers in the Filecoin network will create a new demand for Filecoin and strengthen the ecosystem. Native File API integration will allow a lot of developers to create apps and services that can compete with web 2.0 analogs or competitors. 
+
+## Deliverables
+
+Custom datastore for go-ipfs & go-filecoin that will map UnixFS-v2 IPLD structure with native file APIs for MacOS and Windows. It will support encryption-on-the-fly, so files will be encrypted when stored in other IPFS and Filecoin peers, while being plaintext on the client’s machine.
```

Comment:
Can you describe a bit more how you plan for this to work with both IPFS and Filecoin? As this is the Filecoin devgrants program, we want to make sure that it can support Filecoin, but if we can do both, that's even better.

Reply:
this proposal will be built on top of `go-ipfs` packages such as [go-ipfs-blockstore](https://github.com/ipfs/go-ipfs-blockstore). It is used both in `go-ipfs` and `go-filecoin`.
We will start with `go-ipfs`, while double-checking that all changes are compatible with `go-filecoin`. As soon as MVP will be ready we will integrate it into `go-filecoin`